docs/preview/features/powershell/arm.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Scripts related to ARM templates"
+layout: default
+---
+
+# ARM
+
+## Installation
+
+To have access to the following features, you have to import the module:
+
+```powershell
+PS> Import-Module -Name Arcus.Scripting.ARM
+```
+
+## Inject content in ARM template
+
+In certain scenarios, you have to embed content into an ARM template to deploy it.
+However, the downside of it is that it's buried inside the template and tooling around it might be less ideal - An example of this is OpenAPI specifications you'd want to deploy.
+By using this script, you can inject external files inside your ARM template.
+
+| Parameter | Mandatory | Description                                                         |
+| --------- | --------- | ------------------------------------------------------------------- |
+| `Path`    | no        | The file path to the ARM template to inject the external files into |
+
+**Usage**
+Annotating content to inject:
+
+```json
+{
+    "type": "Microsoft.ApiManagement/service/apis",
+    "name": "[concat(parameters('ApiManagement.Name'),'/', parameters('ApiManagement.Api.Name'))]",
+    "apiVersion": "2019-01-01",
+    "properties": {
+        "subscriptionRequired": true,
+        "path": "demo",
+        "value": "${ FileToInject='.\\..\\openapi\\api-sample.json', InjectAsJsonObject}$",
```

Comment:
It was this example that I was talking about. When I use the file path directly it replaces it, but not with this extra `FileToInject=` part.

Reply:
Hm that's odd. I use that on some projects 🤔

Reply:
Just took a copy of this script and ran it locally (both as a standalone script and as part of the module) and both versions seem to be working for me.
Values that have been tested:
- `"value": "${ FileToInject='.\\..\\openapi\\warehouse.json'}$"`
- `"value": "${ FileToInject='.\\..\\openapi\\warehouse.json', InjectAsJsonObject}$"`
- `"value": "${ FileToInject='.\\..\\api-gateway\\apis\\warehouse\\All-Operations.xml', EscapeJson, ReplaceSpecialChars }$"`

Not entirely sure why this isn't working for you @stijnmoreels.

What didn't work for me was:
- `"value": "${ FileToInject='C:\\Temp\\customer\\openapi\\warehouse.json', InjectAsJsonObject}$"`
As it resulted into a duplication of folder-structure in the filepath.
![image](https://user-images.githubusercontent.com/16165565/88663585-2330ba80-d0dc-11ea-8dd0-ac1933319cf7.png)

Reply:
Also performed the same test with the files provided in `src\Arcus.Scripting.Tests.Integration\Files` and seems to work as expected as well.

Reply:
Which makes me wonder - Can we add an automated test for it to verify it still works fine?

Reply:
Ok, I changed the path in the test file to the path structure you use in hte examples, and now the automated test works.